### ISSUE TITLE
Recordings cut short

### ISSUE THREAD

**Issue by aza547** [aza547/wow-recorder#64]
Suspect zone_change handling problem.

> 1. Recording starts over after 10 minutes (Sylvanas fight is long) and only saves the rest of the fight. Total clip length was 1:44 instead of 11:44.
> 2. Recording the whole fight.
> 3. The rec button I imagine. I didn't have the application open during but all the other fights were recorded correctly.
> 4. Version 2.5.2, I downloaded the application today
[WarcraftRecorder-2022-08-31.log](https://github.com/aza547/wow-recorder/files/9464668/WarcraftRecorder-2022-08-31.log)

**Comment by aza547** [comment]
Suspect promise rejection in logs is a symptom of repeatedly starting OBS because the buffer recording restart kicks in during a real recording. 

> [2022-08-31 21:16:56.498] [error] (node:13012) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)

**Comment by aza547** [comment]
https://drive.google.com/file/d/1GbewXG_BtfF3LgjSrs53hILgkWGLDiMz/view?usp=sharing

**Comment by aza547** [comment]
Probably not a zone change problem. For some reason the cutting took nearly 3 mins. Something must have gone wrong with the async logic as a result?

**Comment by aza547** [comment]
another person, same problem

> Issue 1: Constant beeping during videos. 
> Issue 2: Recordings of KT and Sylvanas pulls are not working. It says the correct pull time on the small thumbnail but actual playtime is wrong. Only showing like 2min 50sec of a 9 min 40 pull and such

[WarcraftRecorder-2022-08-31 (3).log](https://github.com/aza547/wow-recorder/files/9465325/WarcraftRecorder-2022-08-31.3.log)

**Comment by aza547** [comment]
Another instance of this on different bosses:

![image](https://user-images.githubusercontent.com/32389325/188018651-ea633c02-1e68-4086-9cdb-e99461fc521e.png)
[2022-09-01_19-45-02.txt](https://github.com/aza547/wow-recorder/files/9473585/2022-09-01_19-45-02.txt)
[WarcraftRecorder-2022-09-01 (1).log](https://github.com/aza547/wow-recorder/files/9473586/WarcraftRecorder-2022-09-01.1.log)

**Comment by aza547** [comment]
```
[2022-09-01 20:26:20.222] [info]  Recorder: Start recording buffer
[2022-09-01 20:26:20.223] [debug] OBS is now recording...
[2022-09-01 20:26:20.223] [debug] Started?
[2022-09-01 20:26:20.314] [debug] Started signalInfo.type: recording (expected: "recording")
[2022-09-01 20:26:20.315] [debug] Started signalInfo.signal: start (expected: "start")
[2022-09-01 20:26:20.315] [debug] Started!
[2022-09-01 20:27:11.951] [info]  Recorder: Start recording
[2022-09-01 20:30:09.604] [info]  Recorder: Restart recording buffer
[2022-09-01 20:30:09.606] [debug] OBS stopping recording...
[2022-09-01 20:30:09.637] [debug] On stop signalInfo.type: recording (expected: "recording")
[2022-09-01 20:30:09.637] [debug] On stop signalInfo.signal: stopping (expected: "stopping")
[2022-09-01 20:30:09.826] [debug] After stop signalInfo.type: recording (expected: "recording")
[2022-09-01 20:30:09.827] [debug] After stop signalInfo.signal: stop (expected: "stop")
[2022-09-01 20:30:09.827] [debug] Stopped!
[2022-09-01 20:30:11.609] [info]  Deleting: C:/Users/alexs/AppData/Local/Temp/WarcraftRecorder/2022-09-01 20-25-11.mp4
[2022-09-01 20:30:11.638] [debug] OBS is now recording...
[2022-09-01 20:30:11.638] [debug] Started?
[2022-09-01 20:30:11.724] [debug] Started signalInfo.type: recording (expected: "recording")
[2022-09-01 20:30:11.725] [debug] Started signalInfo.signal: start (expected: "start")
[2022-09-01 20:30:11.725] [debug] Started!
[2022-09-01 20:32:57.934] [info]  Recorder: Stop recording after 15 seconds
[2022-09-01 20:32:57.934] [info]  Recorder: {"name":"name","category":"Raids","encounterID":2433,"duration":361,"result":true}
[2022-09-01 20:33:12.936] [debug] OBS stopping recording...
[2022-09-01 20:33:12.968] [debug] On stop signalInfo.type: recording (expected: "recording")
[2022-09-01 20:33:12.968] [debug] On stop signalInfo.signal: stopping (expected: "stopping")
[2022-09-01 20:33:13.158] [debug] After stop signalInfo.type: recording (expected: "recording")
[2022-09-01 20:33:13.158] [debug] After stop signalInfo.signal: stop (expected: "stop")
[2022-09-01 20:33:13.158] [debug] Stopped!
[2022-09-01 20:33:15.039] [info]  Video start time was:  -180
[2022-09-01 20:33:15.040] [info]  Avoiding error by not cutting video
[2022-09-01 20:33:15.040] [info]  Ready to cut video.
[2022-09-01 20:33:15.040] [info]  Initial duration: 181.217 Desired duration: 361 Calculated start time: 0
[2022-09-01 20:33:15.862] [info]  FFmpeg cut video succeeded
[2022-09-01 20:33:15.961] [info]  Deleting: C:/Users/alexs/AppData/Local/Temp/WarcraftRecorder/2022-09-01 20-26-20.mp4
```

Thoughts:
- Looks like recording buffer restart kicked in mid-recording.
- We are meant to cancel the time on the call to recorder.start()
- That's not happened for some reason. Not sure why.

**Comment by computerchipdotbiz** [comment]
This is happening to me also, the only full BG video I got was the very first BG I did, and after that, it only captures pretty much the last few mins or second of the BG. 

Let me know if you need any additional info or logs, and thank you for making this app.

**Comment by computerchipdotbiz** [comment]
https://user-images.githubusercontent.com/5994477/188044430-92e84788-bc79-443f-bce3-66215e25629a.mp4

**Comment by aza547** [comment]
Should be fixed in 2.6.0. Closing.
